docs/index.bs

```diff
@@ -3001,21 +3006,23 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
 
       1. Else if |request| is a <a>subresource request</a>, then:
           1. If |client|'s <a>active service worker</a> is non-null, set |registration| to |client|'s <a>active service worker</a>'s <a>containing service worker registration</a>.
-          1. Else, return null.
+          1. Else, return null and |requestBody|.
       1. Let |activeWorker| be |registration|'s <a>active worker</a>.
       1. Let |shouldSoftUpdate| be true if any of the following are true, and false otherwise:
           * |request| is a [=non-subresource request=].
           * |request| is a [=subresource request=] and |registration| is [=stale=].
       1. If the result of running the [=Should Skip Event=] algorithm with "fetch" and |activeWorker| is true, then:
           1. If |shouldSoftUpdate| is true, then [=in parallel=] run the [=Soft Update=] algorithm with |registration|.
-          1. Return null.
+          1. Return null and |requestBody|.
       1. If |activeWorker|'s <a>state</a> is "`activating`", wait for |activeWorker|'s <a>state</a> to become "`activated`".
       1. If the result of running the [=Run Service Worker=] algorithm with |activeWorker| is *failure*, then set |handleFetchFailed| to true.
       1. Else:
           1. Set |workerRealm| to the [=relevant realm=] of the |activeWorker|'s [=service worker/global object=].
           1. Set |eventHandled| to [=a new promise=] in |workerRealm|.
           1. [=Queue a task=] |task| to run the following substeps:
               1. Let |e| be the result of <a>creating an event</a> with {{FetchEvent}}.
+              1. If |requestBody| is not null, then:
+                  1. Set |requestBody|'s [=Body/stream=] to the result of [=ReadableStream/create a proxy|creating a proxy=] for |requestBody|'s [=Body/stream=].
```

Comment:
Shouldn't this happen much earlier? We need to essentially transfer the request body back. If we just return it there are no side effects for the service worker side and there need to be when it's backed by a stream, right?

Reply:
Which point do you think is the best?

Do you mean [transferring stream](https://github.com/whatwg/streams/issues/276) by "transfer"? I think what's needed is a bit different from that transferring as 

 - We don't keep chunk boundary.
 - Chunks other than Uint8Arrays are rejected.

I think we need to define a means to create such a Body and/or ReadableStream in the fetch spec, but I don't want it to block streaming upload - it is an existing problem for Response.body, too.

Reply:
It became a problem when we added streams. I think now would be a good time to solve it.

And what I mean is that if body is a stream with a null source, service workers needs to transfer ownership back to fetch, such that invoking `req.text()` will start failing afterwards. Presumably this applies to all cases where we return null.

Reply:
Ah, thank you. What do you think about the latest one then?